### PR TITLE
[FEATURE] plugin system 🎉

### PR DESCRIPTION
**Pull request by duivvv**
## Added

You can provide an extra array of functions (= plugin) to manipulate your data after receiving it.

* `plugins`: (Array) Array of functions that receive data and return (manipulated) data.

A plugin function has the signature `data => data`

You can provide them in 2 ways.

* `root` level: passed as an extra argument when creating the root object, used for `all routes`
* `action` level: per route, only triggered on the `provided action` (second argument for the api call)

If you provide plugins at `root level` and at `action level` they are merged into one plugins array in that order

```js
import API from '@teamleader/api';
import { camelCase, normalize } from '@teamleader/api'; // exported as named exports

const { users } = API({
  getAccessToken: () => 'thisisatoken', // async or sync function
  plugins: [camelCase], // at root level
});

// own plugin
const addMeta = data => ({ ...data, meta: { size: data.length } });

const init = async () => {
  // (options, plugins)
  const me = await users.me(undefined, [normalize, addMeta]); // at action level
  console.log(me);
};
```

Both camelCase and normalize are exported as named exports for convenience.

Following the example above:

### camelCase

Recursively convert object keys to camelCase.

data in:

```js
{
  data: [
    {
      id: '8799873',
      user_id: '6979873',
      user_info: {
        first_name: 'Geoffrey',
      },
    },
    {
      id: '3287687',
      user_id: '298034',
      user_info: {
        first_name: 'John',
      },
    },
  ];
}
```

data out:

```js
{
  data: [
    {
      id: '8799873',
      userId: '6979873',
      userInfo: {
        firstName: 'Geoffrey',
      },
    },
    {
      id: '3287687',
      userId: '298034',
      userInfo: {
        firstName: 'John',
      },
    },
  ];
}
```

### normalize

Normalize data structure using byId for easy access and linking.

data in:

```js
{
  data: [
    {
      id: '8799873',
      userId: '6979873',
      userInfo: {
        firstName: 'Geoffrey',
      },
    },
    {
      id: '3287687',
      userId: '298034',
      userInfo: {
        firstName: 'John',
      },
    },
  ];
}
```

data out:

```js
{
  byId: {
    '8799873': {
      id: '8799873',
      userId: '6979873',
      userInfo: {
        firstName: 'Geoffrey',
      },
    },
    '3287687': {
      id: '3287687',
      userId: '298034',
      userInfo: {
        firstName: 'John',
      },
    }
  }
}
```

### CODE REVIEW

File: src/plugins/camelCase.js [hunk 1]
```diff
@@ -0,0 +1,2 @@
+import { camelizeKeys } from 'humps';
+export default data => camelizeKeys(data);
```
Comment (lowiebenoot) [inline src/plugins/camelCase.js]: I guess you can just export camlizeKeys directly?
`export default camelizekeys;`

Comment (JorenSaey) [reply src/plugins/camelCase.js]: Yeah but I like the separation between the used package and the implemented function. Definitely when in the future another package is used. (Just my opninion though)

Comment (duivvv) [reply src/plugins/camelCase.js]: I think he just meant

```
import { camelizeKeys } from 'humps';
export camelizeKeys;

```

@lowiebenoot ?

Comment (lowiebenoot) [reply src/plugins/camelCase.js]: Yes (but `export default`)

Comment (duivvv) [reply src/plugins/camelCase.js]: exactly 🤐

**Pull request merged**